charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -194,9 +200,14 @@ def get_relation_data(self) -> dict:
         # Get relation data from remote app
         relation_data = relation.data[relation.app]
 
+        # Convert string gateway_up back to boolean, defaulting to True if it does not exist.
```

Comment:
Where are you converting back to boolean?

Reply:
here it is, = to comparison 
`gateway_up = gateway_up.lower() == "true"`
I also spend a bit to understand yesterday :)